module/Api/src/Domain/CommandHandler/Messaging/Message/Create.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dvsa\Olcs\Api\Domain\CommandHandler\Messaging\Message;
+
+use Dvsa\Olcs\Api\Domain\Command\Result;
+use Dvsa\Olcs\Api\Domain\CommandHandler\AbstractCommandHandler;
+use Dvsa\Olcs\Api\Domain\Exception\BadRequestException;
+use Dvsa\Olcs\Api\Domain\Repository\Conversation as ConversationRepo;
+use Dvsa\Olcs\Api\Domain\Repository\Message as MessageRepo;
+use Dvsa\Olcs\Api\Domain\Repository\MessageContent as MessageContentRepo;
+use Dvsa\Olcs\Api\Domain\ToggleAwareTrait;
+use Dvsa\Olcs\Api\Domain\ToggleRequiredInterface;
+use Dvsa\Olcs\Api\Entity\Messaging\MessagingContent;
+use Dvsa\Olcs\Api\Entity\Messaging\MessagingConversation;
+use Dvsa\Olcs\Api\Entity\Messaging\MessagingMessage;
+use Dvsa\Olcs\Api\Entity\System\FeatureToggle;
+use Dvsa\Olcs\Transfer\Command\CommandInterface;
+use Dvsa\Olcs\Transfer\Command\Messaging\Message\Create as CreateMessageCommand;
+
+final class Create extends AbstractCommandHandler implements ToggleRequiredInterface
+{
+    use ToggleAwareTrait;
+
+    public const EXCEPTION_MESSAGE_UNABLE_TO_ADD_MESSAGE_TO_CLOSED_ARCHIVED_CONVERSATION = 'Unable to create message on conversations that are closed or archived';
+
+    protected $toggleConfig = [FeatureToggle::MESSAGING];
+    protected $extraRepos = [
+        ConversationRepo::class,
+        MessageRepo::class,
+        MessageContentRepo::class,
+    ];
+
+    public function handleCommand(CommandInterface $command)
+    {
+        assert($command instanceof CreateMessageCommand);
```

Comment:
S: Remove assert. Bit of a discussion on one of Saul's tickets a few days ago. You can turn them off in PHP.ini so can't be relied on (PHP docs say only use for debugging/testing).